serde/src/de/impls.rs

```diff
@@ -295,6 +297,33 @@ impl Deserialize for String {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+#[cfg(feature = "std")]
+impl Deserialize for Box<CStr> {
```

Comment:
No, keep the Serialize impl. It's fine to support serializing a type without supporting deserializing it.